### DISCUSSION TITLE
osu! lazer refuses to open on MacOS 14.5 (Sonoma)

### DISCUSSION THREAD

**Discussion by doctorisyes** [ppy/osu#28917] [Q&A] [answered]
### Type

Crash to desktop

### Bug description

I didn't play osu! for a week. I came back and tried to open it and it jumped on my dock but that was it. It never left a crash report. I tried tracking logs in terminal. Nothing. I tried deleting osu's files in application support, reinstalling it, trying the newest and older versions, nothing.

I am on the latest MacOS version. I have tried it in safe mode in case of a software conflict and ensured all of the permissions osu needs have been enabled in settings. I have tried creating another user, and still the same issue. I've tried deleting framework.ini. Nothing.

I tried side loading the iOS version of osu! onto my Mac and that did work if that is any help.

There's no crash report at all to be found, just says terminated when I find it in activity monitor. There's no errors apparently. Just refuses to open.

### Screenshots or videos

<img width="564" alt="Screenshot 2024-07-13 at 16 08 37" src="https://github.com/user-attachments/assets/99563cc3-363e-4183-8f61-4e2699fb8604">

https://github.com/user-attachments/assets/5e4cf1ef-64fe-4907-a033-4349b18bdf95

### Version

2024.625.2

### Logs

I went into the logs folder and the last log generated was from the 6th July. This is the most that I could get as a log. Likeliness is these are completely irrelevant:

[1720258505.auth.log](https://github.com/user-attachments/files/16227725/1720258505.auth.log)
[1720258505.database.log](https://github.com/user-attachments/files/16227726/1720258505.database.log)

I have absolutely no way to show what was going on. I tried to use terminal show what was going on with this command (log stream --predicate 'process == "osu!"' --info)  and I just got this: 

2024-07-14 14:49:50.851995+0100 0x29bb4    Default     0x0                  2561   0    osu!: (libxpc.dylib) [com.apple.xpc:connection] [0x6000034244b0] activating connection: mach=true listener=false peer=false name=com.apple.cfprefsd.agent

EDIT:
I have now become extremely sad without osu!. So I have now reset my MacBook, and that has seemed to stop the issue. Only problem is I accidentally forgot to backup my school work in the process, but that's my fault lol. At least I can play osu now!

**Comment by frenzibyte** [comment]
Do you have any other application running? If so, try closing all of them and run the game once again.

EDIT: I noticed you've tried to run this on safe mode, so I guess that rules this question out.

**Comment by smoogipoo** [comment]
Have you downloaded the correct version of the game (Apple Silicon vs Intel)?

**Reply by doctorisyes** [reply]
yes. I tried the apple silicon version. that's my usual version.

**Comment by frenzibyte** [comment]
Did you fix this issue? You've marked this thread as "resolved".

**Comment by so3ur** [comment]
same here, i have no idea how this would correlate, but have you also deactivated SIP? I remember osu!lazer working fine before i toyed around with that for playcover

**Comment by sealvessel** [comment]
It works if you don't disable system integrity protection. It's probably some anticheat mechanism.

**Comment by doctorisyes** [comment, marked as answer]
I found the issue. after all this time. 

In the past I had created a network service on the loopback interface. This made osu lazer unable to listen to ports on localhost and I assume that caused the issue. I know this is probably what caused it, because I eventually re installed my entire OS and started from scratch and osu! worked again, but when I interfered with my network settings again like I did last time it messed up OSU immediately. it must be that.

**Reply by peppy** [reply]
Can you show what you did to break it?

**Reply by doctorisyes** [reply]
sudo networksetup -createnetworkservice AdHoc lo0
sudo networksetup -setmanual AdHoc 192.168.1.88 255.255.255.255

I would have never guessed this would interfere with osu! but it did. However, OSU wasn't alone with this problem. Steam didn't open after creating this. I was just trying to make a server that could be used on a wifi-connection without any actual internet connection.

**Reply by peppy** [reply]
Thanks. I don't think anyone else is ever going to run into this so I'll just leave it here 🙈
